atom/browser/atom_browser_client.cc

```diff
@@ -100,20 +100,29 @@ bool AtomBrowserClient::ShouldCreateNewSiteInstance(
 
   int process_id = current_instance->GetProcess()->GetID();
   if (!IsRendererSandboxed(process_id)) {
-    if (!RendererUsesNativeWindowOpen(process_id)) {
+    auto web_contents =
+        content::WebContents::FromRenderFrameHost(render_frame_host);
+    if (!WebContentsPreferences::UsesNativeWindowOpen(web_contents)) {
       // non-sandboxed renderers without native window.open should always create
       // a new SiteInstance
       return true;
     }
-    auto web_contents =
-        content::WebContents::FromRenderFrameHost(render_frame_host);
+
     if (!ChildWebContentsTracker::IsChildWebContents(web_contents)) {
       // Root WebContents should always create new process to make sure
       // native addons are loaded correctly after reload / navigation.
       // (Non-root WebContents opened by window.open() should try to
       //  reuse process to allow synchronous cross-window scripting.)
       return true;
     }
+
+    // In a non-sandboxed renderer with native window open we should
+    // reuse the same site to allow cross-window scripting.  We do
+    // not need to check urls / domains as native window open logic
+    // handles cross site scripting protection.
+    if (WebContentsPreferences::UsesNativeWindowOpen(web_contents)) {
```

Comment:
Isn't this `if` guaranteed to always be `true` since we didn't enter the `if (!WebContentsPreferences::UsesNativeWindowOpen(web_contents))` block above?